system/jlib/jsocket.cpp

```diff
@@ -2939,7 +2947,10 @@ bool getInterfaceIp(IpAddress &ip,const char *ifname)
 #else
             IpAddress iptest((inet_ntoa(((struct sockaddr_in *)&item->ifr_addr)->sin_addr)));
 #endif
-            if (ioctl(fd, SIOCGIFFLAGS, item) < 0)
+            struct ifreq flags;
```

Comment:
Can't safely use same buffer for the flags result - it may overwrite the next entry in the list in osx case.

Reply:
good find